lighthouse-core/lib/tracehouse/main-thread-tasks.js

```diff
@@ -262,10 +262,42 @@ class MainThreadTasks {
             // It's ending at traceEndTs, it means we were missing the end event. We'll truncate it to the parent.
             nextTask.endTime = currentTask.endTime;
             nextTask.duration = nextTask.endTime - nextTask.startTime;
+          } else if (
+            nextTask.startTime - currentTask.startTime < 1000 &&
+            !currentTask.children.length
+          ) {
+            // The parent started less than 1ms before the child, we'll let it slide by swapping the two,
+            // and increasing the duration of the parent. Below is an artistic rendition of this situation.
```

Comment:
this is the rendition *after* the swap, right?

Reply:
errr, kind of. once we fix everything `nextTask` and `currentTask` labels will be switched so the diagram looks correct. at this exact moment in the code there is no heirarchy relationship whatsoever between nextTask and currentTask this diagram is attempting to take the true structure of things and apply the variable labels we currently have for them.

open to whatever explanation you think fits best and doesn't confuse me too :)